aiida_registry/fetch_metadata.py

```diff
@@ -31,6 +32,22 @@
 
 GITHUB_TOKEN = os.environ["GITHUB_TOKEN"]
 
+def get_last_fetched_version(plugin_name):
```

Comment:
This function has a static, plugin-independent part (reading and parsing the file), and a per-plugin part.

I suggest you move the static part outside this function.
You can either use a singleton variable to store the content or use a separate function with functools.lru_cache

Reply:
Please also document the plugin-specific function, at least in terms of return type